### PR TITLE
Add dns::powerdns options

### PR DESCRIPTION
**Pull request by ekohl**
Closes GH-242

### CODE REVIEW

File: manifests/plugin/dns/powerdns/params.pp [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
-  $mysql_password  = cache_data('foreman_cache_data', 'smart_proxy_dns_powerdns_password', random_password(16))
-  $mysql_database  = 'pdns'
-  $manage_database = false
-  $pdnssec         = 'pdnssec'
+  $backend               = 'mysql'
+  $mysql_hostname        = 'localhost'
+  $mysql_username        = 'pdns'
+  $mysql_password        = cache_data('foreman_cache_data', 'smart_proxy_dns_powerdns_password', random_password(16))
+  $mysql_database        = 'pdns'
+  $postgresql_connection = ''
```
Comment (brandonweeks) [inline manifests/plugin/dns/powerdns/params.pp]: Tests are failing on this:
>manifests/plugin/dns/powerdns/params.pp:7:WARNING: variable assigned to the empty string

**Comment by mmoll** [comment]
@ekohl what's the status here?

**Comment by ekohl** [comment]
Updated

File: manifests/plugin/dns/powerdns.pp [hunk 2]
```diff
@@ -4,31 +4,40 @@
 #
 # === Parameters:
 #
-# $mysql_hostname::  MySQL server hostname
+# $backend::               The backend to select, either mysql or postgres.
```
Comment (domcleal) [inline manifests/plugin/dns/powerdns.pp]: `postgresql` according to the plugin source code/README

File: templates/plugin/dns_powerdns.yml.erb [hunk 3]
```diff
@@ -1,6 +1,12 @@
 ---
+<% backend = scope.lookupvar('::foreman_proxy::plugin::dns::powerdns::backend') -%>
+:powerdns_backend: "<%= backend %>"
+<% if backend == 'mysql' -%>
 :powerdns_mysql_hostname: "<%= scope.lookupvar('::foreman_proxy::plugin::dns::powerdns::mysql_hostname') %>"
 :powerdns_mysql_username: "<%= scope.lookupvar('::foreman_proxy::plugin::dns::powerdns::mysql_username') %>"
 :powerdns_mysql_password: "<%= scope.lookupvar('::foreman_proxy::plugin::dns::powerdns::mysql_password') %>"
 :powerdns_mysql_database: "<%= scope.lookupvar('::foreman_proxy::plugin::dns::powerdns::mysql_database') %>"
+<% elsif backend == 'postgres' -%>
```
Comment (domcleal) [inline templates/plugin/dns_powerdns.yml.erb]: ditto, `postgresql` (and throughout tests etc)

**Comment by ekohl** [comment]
Tests will fail on https://github.com/fiddyspence/puppetlint-variablecase/pull/6

**Comment by ekohl** [comment]
Rebased

File: manifests/plugin/dns/powerdns.pp [hunk 4]
```diff
@@ -4,31 +4,40 @@
...
+  $mysql_username        = $::foreman_proxy::plugin::dns::powerdns::params::mysql_username,
+  $mysql_password        = $::foreman_proxy::plugin::dns::powerdns::params::mysql_password,
+  $mysql_database        = $::foreman_proxy::plugin::dns::powerdns::params::mysql_database,
+  $postgresql_connection = $::foreman_proxy::plugin::dns::powerdns::params::postgresql_connection,
+  $manage_database       = $::foreman_proxy::plugin::dns::powerdns::params::manage_database,
+  $pdnssec               = $::foreman_proxy::plugin::dns::powerdns::params::pdnssec,
 ) inherits foreman_proxy::plugin::dns::powerdns::params {
   validate_bool($manage_database)
-  validate_string($mysql_hostname, $mysql_username, $mysql_password, $mysql_database, $pdnssec)
+  validate_re($backend, 'mysql|postgres', 'Invalid backend: choose mysql or postgresql')
```
Comment (mmoll) [inline manifests/plugin/dns/powerdns.pp]: postgres vs. postgresql

Comment (ekohl) [inline manifests/plugin/dns/powerdns.pp]: argh, missed one ... I guess it worked because the regex also matches it but will make it stricter.

**Comment by domcleal** [comment]
Merged, thanks @ekohl!

**Pull request merged**